### DISCUSSION TITLE
Helm controller doesn't seem to support merging text into values.yaml as literals

### DISCUSSION THREAD

**Discussion by ntjn** [fluxcd/flux2#3122] [Proposals]
(I've been using Flux for almost a year now, stumbled upon a chart that introduced this cornercase.)

## The problem I need to solve
The chart I want to manage by GitOps requires an attribute of _values.yaml_ to be specified as text literal (I'll show what I mean by that). This data is sensitive, which I would normally merge from a secret or configmap. Let's assume we are using a configmap now.

### Merging data with `valuesFrom`
Let _p.yaml_ look like this:
  
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: p
      namespace: p
    data:
      p: |
      sensitive-information: whimsicality

And _values.yaml_ like this:
    
    q:
      r: null

Now I want to merge p (into `q.r`) with `valuesFrom` (HelmRelease CRD), and the merge renders the following _values.yaml_:
    
    q:
      r:
        sensitive-information: whimsicality

However, what I want is merging the data as literal (it's not a yaml). So this would like as follows:

    q:
      r: |
        sensitive-information: whimsicality

Helm CLI supports doing so with the `--set-file` flag, but the Helm controller doesn't.

## Potential solution
Adding a new flag to the [ValuesReference](https://fluxcd.io/flux/components/helm/api/#helm.toolkit.fluxcd.io/v2beta1.ValuesReference) that indicates merging as literal.

Do you want to support that? If so, then I gladly come up with a solution.

**Comment by airtonix** [comment]
I'm seeing the opposite behaviour: 

https://github.com/fluxcd/flux2/discussions/3163

**Reply by ntjn** [reply]
I see, patch seems to solve the problem. Going to try this out.
